test/crowdsale/TimedCrowdsale.test.js

```diff
@@ -73,5 +73,67 @@ contract('TimedCrowdsale', function ([_, investor, wallet, purchaser]) {
         await shouldFail.reverting(this.crowdsale.buyTokens(investor, { value: value, from: purchaser }));
       });
     });
+
+    describe('extending closing time', function () {
+      it('should not reduce duration', async function () {
+        (await this.crowdsale.isOpen()).should.equal(false);
+        await shouldFail.reverting(this.crowdsale.send(value));
+
+        // Same date
+        await shouldFail.reverting(this.crowdsale.extendTime(this.closingTime));
+
+        // Prescending date
+        const newClosingTime = this.closingTime.sub(time.duration.seconds(1));
+        await shouldFail.reverting(this.crowdsale.extendTime(newClosingTime));
+      });
+
+      context('before crowdsale start', function () {
+        beforeEach(async function () {
+          (await this.crowdsale.isOpen()).should.equal(false);
+          await shouldFail.reverting(this.crowdsale.send(value));
+        });
+
+        it('it extends end time', async function () {
+          const newClosingTime = this.closingTime.add(time.duration.days(1));
+          const { logs } = await this.crowdsale.extendTime(newClosingTime);
+          expectEvent.inLogs(logs, 'TimedCrowdsaleExtended', {
+            prevClosingTime: this.closingTime,
+            newClosingTime: newClosingTime,
+          });
+          (await this.crowdsale.closingTime.call()).should.be.bignumber.equal(newClosingTime);
+        });
+      });
+
+      context('after crowdsale start', function () {
+        beforeEach(async function () {
+          await time.increaseTo(this.openingTime);
+          (await this.crowdsale.isOpen()).should.equal(true);
+          await this.crowdsale.send(value);
+        });
+
+        it('it extends end time', async function () {
+          const newClosingTime = this.closingTime.add(time.duration.days(1));
+          const { logs } = await this.crowdsale.extendTime(newClosingTime);
+          expectEvent.inLogs(logs, 'TimedCrowdsaleExtended', {
+            prevClosingTime: this.closingTime,
+            newClosingTime: newClosingTime,
+          });
+          (await this.crowdsale.closingTime.call()).should.be.bignumber.equal(newClosingTime);
+        });
+      });
+
+      context('after crowdsale end', function () {
+        beforeEach(async function () {
+          await time.increaseTo(this.afterClosingTime);
+          (await this.crowdsale.isOpen()).should.equal(false);
+          await shouldFail.reverting(this.crowdsale.send(value));
+        });
+
+        it('it reverts', async function () {
+          const newClosingTime = this.closingTime.add(time.duration.days(1));
```

Comment:
```suggestion
          const newClosingTime = this.afterClosingTime.add(time.duration.days(1));
```

You've just increased time to `afterClosingTime`, so the new endtime should be after it. We could also use `time.latest()`.

Reply:
Sure